clang/lib/Sema/SemaBounds.cpp

```diff
@@ -3920,13 +3936,26 @@ namespace {
     // whose observed bounds are unknown after checking the top-level CFG
     // statement St.
     //
-    // State contians information that is used to provide more context in
+    // State contains information that is used to provide more context in
     // the diagnostic messages.
     void DiagnoseUnknownObservedBounds(Stmt *St, const VarDecl *V,
                                        BoundsExpr *DeclaredBounds,
                                        CheckingState State) {
-      S.Diag(St->getBeginLoc(), diag::err_unknown_inferred_bounds)
-        << V << St->getSourceRange();
+      // Blame the assignment in St that causes V's observed bounds to be
+      // unknown if we can find it in map BlameAssignments.  Otherwise, blame
+      // the whole St in the error message.
+      SourceLocation Loc = St->getBeginLoc();
+      SourceRange SrcRange = St->getSourceRange();
+      auto BDCType = Sema::BoundsDeclarationCheck::BDC_Statement;
```

Comment:
This could miss the case where the bounds of `V` are unknown after an initialization, e.g.

```
array_ptr<int> unknown_src : bounds(unknown);
// Bounds of arr are unknown after initialization
array_ptr<int> arr : count(1) = unknown_src;
```